File: JavaScript/1-TwoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

function twoSum(nums, target) {
    const obj = {};
    for (let i = 0; i < nums.length; i++) {
        if (obj[target - nums[i]] !== undefined) {
            return [i, obj[target - nums[i]]];
        } else {
            obj[nums[i]] = i;
        }
    }
}

File: Python/twosum.py
class Solution(object):
    def twoSum(self, nums, target):
        nums.sort()
        result = []
        slen = len(nums)
        for i in range(slen - 1):
            for j in range(i + 1, slen):
                if nums[i] + nums[j] == target:
                    result.append(i)
                    result.append(j)
                    return result

if __name__ == '__main__':
    souletion = Solution()
    res = souletion.twoSum([3, 2, 4], 6)
    print res

